libraries/lib-wave-track/WaveTrack.cpp

```diff
@@ -2969,8 +2969,7 @@ const WaveClip* WaveTrack::GetClipAtTime(double time) const
    const auto clips = SortedClipArray();
    auto p = std::find_if(
       clips.rbegin(), clips.rend(), [&](const WaveClip* const& clip) {
-         return time >= clip->GetPlayStartTime() &&
-                time <= clip->GetPlayEndTime();
+         return clip->WithinPlayRegion(time);
```

Comment:
A small change.  But I remember some of the surprises in getting effects figured out, were the unintended consequences of changing `<` to `<=` tests in certain `WaveTrack` and `WaveClip` member functions that were more widely used.

Are there such surprises here?

I did review all uses of `GetClipTime` and now, I think not.

Reply:
When I changed the boundary check in the first place, it was to redefine clip intervals as standard closed-begin, open-end. Consistently, and for adjacent clips, `clipA.GetPlayEndTime() == clipB.GetPlayEndTime()`.

This is a place that went against this rule of thumb that I haven't addressed earlier, and would prevent me from using `GetClipAtTime` in `PasteOne`: there I don't want a clip whose end is at `t0` to interfere.

I also looked at uses of `GetClipAtTime`. It's used in `GetEnvelopeAtTime`. My testing of envelope editing showed that it still worked properly. Will add a QA checkbox.